README.md

```diff
@@ -110,89 +152,65 @@ execute the following in any OCaml top-level:
 ```ocaml
 # #use "topfind";;
 # #require "bap.top";;
-# open Bap.Std;;
 ```
 
-And everything should work just out of box, i.e. it will load all the
-dependencies, install top-level printers, etc.
+## Shell
 
+The BAP main frontend is a command line utility called
+[bap](https://github.com/BinaryAnalysisPlatform/bap-python). You can
+use it to explore the binary, run existing analysis, plugin your own
+behavior, load traces, and many more.
 
-## Using from shell
+To dump a program in various formats use `--dump` option (or its short
+equivalent `-d`). By default it dumps in the IR, but you can choose
+from various other formats. Use `--list-formats` option, to get the
+list. Note, however, that formats are added by plugins, and there are
+plugins lurking somewhere in opam, that may bring your favorite format
+to bap, for example, `bap-piqi` package, brings support for a plethora
+of formats (protobuf, xml, json, to name a few).
 
-Bap is shipped with `bap` utility that can disassemble files, and
-printout dumps in different formats, including plain text, json, dot,
-html. The example of `bap` output is:
+To discover what plugins are currently available, use `--list-plugins`
+option. A short description will be printed for each plugin. You can
+also use `opam search bap` command, to get the information about other
+bap packages.
 
-```
-00000088: sub strcpy(arg_0, arg_1)
-00000151: arg_0 :: u32 = R0
-00000152: arg_1 :: u32 = R1
-0000005f:
-00000063: ZF.1 := R0 = 0x0:32
-00000064: when ZF.1 return LR
-00000065: goto %00000066
-
-00000066:
-00000067: t_614.1 := mem[R1, el]:u8
-00000068: R3.1 := pad:32[t_614.1]
-0000006c: ZF.2 := R3.1 = 0x0:32
-0000006d: when ZF.2 goto %0000006f
-0000006e: goto %00000076
-
-0000006f:
-00000070: R12.1 := R0
-00000071: goto %00000072
-
-00000072:
-0000012f: R1.1 := phi([R1, %0000006f], [R1.3, %00000086])
-00000131: R12.2 := phi([R12.1, %0000006f], [R12.4, %00000086])
-00000133: R2.1 := phi([R2, %0000006f], [R2.4, %00000086])
-00000135: R3.2 := phi([R3.1, %0000006f], [R3.5, %00000086])
-00000137: mem.1 := phi([mem, %0000006f], [mem.4, %00000086])
-00000073: R3.3 := 0x0:32
-00000074: mem.2 := mem.1 with [R12.2, el]:u8 <- low:8[R3.3]
-00000075: return LR
-...
-```
-
-Also we're shipping a `bap-mc` executable that can disassemble
-arbitrary strings and output them in a plethora of formats. Read
-`bap-mc --help` for more information. `bap-byteweight` utility can be
-used to evaluate our `byteweight` algorithm for finding symbols inside
-the binary. It is also a supporting toolkit for byteweight
-infrastructure, it can download, create and install binary signatures,
-used for identification.
+To get information about specific plugin named `<PLUGIN>` use
+`--<PLUGIN>-help` option, e.g., `bap --llvm-help`.
 
+The `bap` utility works with the whole binaries, if you have just few
```

Comment:
, - > ;
no the

The `bap` utility works with whole binaries; if you have just few